Sources/ApolloCodegenLib/Templates/InputObjectTemplate.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+import JavaScriptCore
+
+struct InputObjectTemplate {
+  let graphqlInputObject: GraphQLInputObjectType
+
+  func render() -> String {
+    TemplateString(
+    """
+    struct \(graphqlInputObject.name.firstUppercased): InputObject {
+      private(set) public var dict: InputDict
+
+      init(
+        \(InitializerParametersTemplate())
+      ) {
+        dict = InputDict([
+          \(InputDictInitializerTemplate())
+        ])
+      }
+
+      \(graphqlInputObject.fields.map({ "\(FieldPropertyTemplate($1))" }), separator: "\n\n")
+    }
+    """
+    ).value
+  }
+
+  private func InitializerParametersTemplate() -> TemplateString {
+    TemplateString("""
+    \(graphqlInputObject.fields.map({
+      "\($1.name): \($1.renderType(includeDefault: true))"
+    }), separator: ",\n")
+    """)
+  }
+
+  private func InputDictInitializerTemplate() -> TemplateString {
+    TemplateString("""
+    \(graphqlInputObject.fields.map({ "\"\($1.name)\": \($1.name)" }), separator: ",\n")
+    """)
+  }
+
+  private func FieldPropertyTemplate(_ field: GraphQLInputField) -> String {
+    """
+    var \(field.name): \(field.renderType()) {
+      get { dict[\"\(field.name)\"] }
+      set { dict[\"\(field.name)\"] = newValue }
+    }
+    """
+  }
+}
+
+fileprivate extension GraphQLInputField {
+  func renderType(includeDefault: Bool = false) -> String {
+    "\(type.render())\(isSwiftOptional ? "?" : "")\(includeDefault && hasSwiftNilDefault ? " = nil" : "")"
+  }
+
+  private var isSwiftOptional: Bool {
+    !isNullable && hasSchemaDefault
+  }
+
+  private var hasSwiftNilDefault: Bool {
+    isNullable && !hasSchemaDefault
+  }
+
+  private var isNullable: Bool {
+    switch type {
+    case .nonNull(_): return false
+    default: return true
+    }
+  }
+
+  private var hasSchemaDefault: Bool {
+    switch defaultValue {
+    case .none, .some(nil):
+      return false
+    case let .some(value):
+      guard let value = value as? JSValue else {
+        fatalError("Cannot determine default value for Input field: \(self)")
+      }
+
+      return !value.isUndefined
+    }
+  }
+}
+
+fileprivate extension GraphQLType {
+  func render(containedInNonNull: Bool = false) -> String {
+    switch self {
+    case let .entity(type as GraphQLNamedType),
+      let .enum(type as GraphQLNamedType),
+      let .inputObject(type as GraphQLNamedType):
+
+      return containedInNonNull ? type.name : "GraphQLNullable<\(type.name)>"
+
+    case let .scalar(type as GraphQLNamedType):
+      let typeName = (type.name == "Boolean" ? "Bool" : type.name)
```

Comment:
I wanted to have this come from JS but it looks like the frontend gets [this type from graphql-js](https://github.com/graphql/graphql-js/blob/main/src/type/scalars.ts#L176) and not our bridging/frontend code so here seems like the 'right' place, _could possible be moved somewhere common for the rest of the code to use if needed though._

Reply:
I'm not a giant fan of doing this as a one-liner instead of an if/else - this isn't going into client code so we don't need to be wary of saving lines. 

Might be worth doing it as a `swiftName` extension and hiding the logic in there?

Reply:
See f9eb935ac7eee92446ce76da25b1e5450801310c for the implementation.

Reply:
The alternative to that extension is a simple `typealias` but I'm not sure how I feel about all our generated code being `Boolean` vs. `Bool` - should we care about it?

Reply:
@AnthonyMDev do you have a preference?

Reply:
I think a typealias is way more confusing, personally

Reply:
I generally agree because `typealias` names are usually dissimilar from the origin but in this case I could get comfortable with it given that `Boolean` as a type name is pretty clear.

Reply:
I like the `swiftName` property in the extension. Not a typealias for sure.